python/paddle/fluid/io.py

```diff
@@ -473,79 +478,154 @@ def save_checkpoint(executor,
 
     :param executor
     :param checkpoint_dir
-    :param max_num_checkpoints
-    :param save_interval_secs
+    :param trainer_id
+    :param is_chief
     :param main_program
+    :param max_num_checkpoints
     """
     if checkpoint_dir is None:
-        checkpoint_dir = os.getcwd()
+        raise ValueError("The values of 'checkpoint_dir' should not be None")
+
+    if trainer_args and not isinstance(trainer_args, dict):
```

Comment:
```trainer_args ``` not for the user, just for the developers, currently， ```trainer_args ``` only contains ``` step_id, epoch_id ```， there maybe have more arguments need to be saved in the checkpoint.

Reply:
Please use the clear parameters, it would be confused with other developers, use `step_id` and `epoch_id` or a Class as the configuration parameter.

Reply:
I write two functions to make it more clearly.